cobc/pplex.l

```diff
@@ -260,82 +260,105 @@ MAYBE_AREA_A	[ ]?#?
 	}
 }
 
-^[ ]*">>"[ ]?"PAGE"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"PAGE"	{
 	/* 2002+: listing directive for page eject with optional comment
 	   Note: processed in cobc.c */
 	skip_to_eol ();
 }
 
-^[ ]*">>"[ ]?"LISTING"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"LISTING"	{
 	/* 2002+: listing directive for (de-)activating the listing,
 	   ON implied for empty value
 	   Note: further checks in ppparse.y, processed in cobc.c */
 	BEGIN ON_OFF_DIRECTIVE_STATE;
 	return LISTING_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]?"SOURCE"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"SOURCE"	{
 	/* 2002+: directive for setting source format */
 	BEGIN SOURCE_DIRECTIVE_STATE;
 	return SOURCE_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]?"SET"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"SET"	{
 	/* OpenCOBOL/GnuCOBOL 2.0 extension: MF SET directive in 2002+ style format */
 	BEGIN SET_DIRECTIVE_STATE;
 	return SET_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]?"TURN"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"TURN"	{
 	/* 2002+: directive for (de-)activating exception checks */
 	BEGIN TURN_DIRECTIVE_STATE;
 	return TURN_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]?"IF"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"IF"	{
 	/* 2002+: conditional compilation */
 	BEGIN IF_DIRECTIVE_STATE;
 	return IF_DIRECTIVE;
 }
-^[ ]*">>"[ ]?"ELIF" |
-^[ ]*">>"[ ]?"ELSE-IF"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"ELIF" |
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"ELSE-IF"	{
 	/* OpenCOBOL extension: conditional compilation combined ELSE IF,
 	   2002+ style format */
 	BEGIN IF_DIRECTIVE_STATE;
 	return ELIF_DIRECTIVE;
 }
-^[ ]*">>"[ ]?"ELSE"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"ELSE"	{
 	/* 2002+: conditional compilation */
 	BEGIN ELSE_DIRECTIVE_STATE;
 	return ELSE_DIRECTIVE;
 }
-^[ ]*">>"[ ]?"END-IF"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"END-IF"	{
 	/* 2002+: conditional compilation */
 	BEGIN ENDIF_DIRECTIVE_STATE;
 	return ENDIF_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]?"LEAP-SECOND"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"LEAP-SECOND"	{
 	/* 2002+: more then 60 seconds per minute (currently always set to off),
 	          OFF implied for empty value */
 	BEGIN ON_OFF_DIRECTIVE_STATE;
 	return LEAP_SECOND_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]?"CALL-CONVENTION"	{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"CALL-CONVENTION"	{
 	/* 2002+: convention for CALL/CANCEL */
 	BEGIN CALL_DIRECTIVE_STATE;
 	return CALL_DIRECTIVE;
 }
 
-^[ ]*">>"[ ]*\n		{
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"AREACHECK"	|
+^{MAYBE_AREA_A}[ ]*">>"[ ]?"AREA-CHECK"	{
+	if (cobc_has_areacheck ()) {
+		fprintf (ppout, "#AREACHECK\n");
+	} else {
+		cb_plex_warning (COBC_WARN_FILLER, newline_count,
+				 _("ignoring AREACHECK directive"));
```

Comment:
to make this more useful for the users and nicer for the translators lets switch here and below to something like

```c
	cb_plex_warning (COBC_WARN_FILLER, newline_count,
				 _("ignoring %s directive because of %s"), "AREACHECK", "-fformat=fixed/free");
```

--> a bit more detail and still one msgid added less